src/eq_schema/Group.js

```diff
@@ -1,15 +1,39 @@
+/* eslint-disable camelcase */
 const Block = require("./Block");
 const { getInnerHTML } = require("../utils/HTMLUtils");
+const { filter, isEmpty, reject } = require("lodash");
 
 class Group {
   constructor(id, title, pages, ctx) {
     this.id = `group${id}`;
     this.title = getInnerHTML(title);
-    this.blocks = this.buildBlocks(this.title, pages, ctx);
+    this.blocks = this.buildBlocks(this.title, pages, id, ctx);
+
+    if (!isEmpty(ctx.routingGotos)) {
+      this.filterContext(this.id, ctx);
+      const skipConditions = this.buildSkipConditions(this.id, ctx);
+
+      if (!isEmpty(skipConditions)) {
+        this.skip_conditions = skipConditions;
+      }
+    }
+  }
+
+  filterContext(currentId, ctx) {
+    ctx.routingGotos = filter(
```

Comment:
Sorry, one final comment! `reject` is probably a better fit for what you're trying to do here? 

```js
reject(ctx.routingGotos, { group: currentId })
```